src/theory/sets/normal_form.h

```diff
@@ -44,11 +44,13 @@ class NormalForm {
     }
     else
     {
+      TypeNode elementType = setType.getSetElementType();
       ElementsIterator it = elements.begin();
-      Node cur = nm->mkNode(kind::SINGLETON, *it);
+      Node cur = nm->mkSingleton(elementType, *it);
```

Comment:
This is safe here because `elementType` is extracted from `setType` that is passed as an argument to `elementsToSet` and is used to construct a set the are no elements.